app/javascripts/helpers/web3Service.jsx

```diff
@@ -160,6 +160,122 @@ class Web3Service {
       );
     });
   }
+  async getSlotMachineInfo(slotMachineContract) {
+    const promiseArr = [
+      this.getSlotMachineAvailable(slotMachineContract),
+      this.getSlotMachineBankrupt(slotMachineContract),
+      this.getSlotMachineMaxBet(slotMachineContract),
+      this.getSlotMachineMinBet(slotMachineContract),
+      this.getSlotMachineProviderBalance(slotMachineContract),
+    ];
+    const payload = {};
+    await Promise.all(promiseArr).then(infoObjArr => {
+      infoObjArr.forEach(infoObj => {
+        payload[infoObj.infoKey] = infoObj.infoVal;
+      });
+    });
+    return payload;
+  }
+
+  getSlotMachineAvailable(slotMachineContract) {
+    return new Promise((resolve, reject) => {
+      slotMachineContract.mAvailable((err, mAvailable) => {
+        if (err) {
+          reject(err);
+        } else {
+          if (!mAvailable) reject('This slot is not avaliable!');
+          resolve({ infoKey: 'avaliable', infoVal: mAvailable });
+        }
+      });
+    });
+  }
+
+  getSlotMachineBankrupt(slotMachineContract) {
+    return new Promise((resolve, reject) => {
+      slotMachineContract.mBankrupt((err, mBankrupt) => {
+        if (err) {
+          reject(err);
+        } else {
+          if (mBankrupt) reject('This slot is bankrupted!');
+          resolve({ infoKey: 'bankrupt', infoVal: mBankrupt });
+        }
+      });
+    });
+  }
+
+  getSlotMachineMaxBet(slotMachineContract) {
+    return new Promise((resolve, reject) => {
+      slotMachineContract.mMaxBet((err, mMaxBet) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve({
+            infoKey: 'maxBet',
+            infoVal: this.getEthFromWei(parseInt(mMaxBet.valueOf(), 10)),
```

Comment:
Yes, Wei is a unit for expressing large numbers. So I think it would be better than use parseFloat